editor/frontend/src/ModifyElementTool.ts

```diff
@@ -0,0 +1,195 @@
+    this.setupAccuDraw();
+    super.setupAndPromptForNextAction();
+  }
+
+  protected abstract getGeometryProps(ev: BeButtonEvent, isAccept: boolean): JsonGeometryStream | FlatBufferGeometryStream | undefined;
+  protected abstract getElementProps(ev: BeButtonEvent): GeometricElementProps | undefined;
+
+  protected async doUpdateElement(_props: GeometricElementProps): Promise<boolean> { return false; }
+
+  protected async applyAgendaOperation(ev: BeButtonEvent): Promise<boolean> {
+    const geometry = this.getGeometryProps(ev, true);
+    if (undefined === geometry)
+      return false;
+
+    const elemProps = this.getElementProps(ev);
+    if (undefined === elemProps)
+      return false;
+
+    if ("flatbuffer" === geometry.format)
+      elemProps.elementGeometryBuilderParams = { entryArray: geometry.data };
+    else
+      elemProps.geom = geometry.data;
+
+    return this.doUpdateElement(elemProps);
+  }
+
+  public override async processAgenda(ev: BeButtonEvent): Promise<void> {
+    if (await this.applyAgendaOperation(ev))
+      return this.saveChanges();
+  }
+}
+
+/** @alpha Edit tool base class for updating existing elements that use dynamics to show intermediate results. */
+export abstract class ModifyElementWithDynamicsTool extends ModifyElementTool implements FeatureOverrideProvider {
+  protected _graphicsProvider?: DynamicGraphicsProvider;
+  protected _firstResult = true;
+  protected _agendaAppearanceDefault?: FeatureAppearance;
+  protected _agendaAppearanceDynamic?: FeatureAppearance;
+
+  protected override get wantAccuSnap(): boolean { return true; }
+  protected override get wantDynamics(): boolean { return true; }
+
+  protected agendaAppearance(isDynamics: boolean): FeatureAppearance {
+    if (isDynamics) {
+      if (undefined === this._agendaAppearanceDynamic)
+        this._agendaAppearanceDynamic = FeatureAppearance.fromTransparency(0.0);
+
+      return this._agendaAppearanceDynamic;
+    }
+
+    if (undefined === this._agendaAppearanceDefault)
+      this._agendaAppearanceDefault = FeatureAppearance.fromTransparency(0.9);
+
+    return this._agendaAppearanceDefault;
+  }
+
+  protected get wantAgendaAppearanceOverride(): boolean { return false; }
+
+  public addFeatureOverrides(overrides: FeatureSymbology.Overrides, _vp: Viewport): void {
+    if (this.agenda.isEmpty)
+      return;
+
+    const appearance = this.agendaAppearance(false);
+    this.agenda.elements.forEach((elementId) => overrides.override({ elementId, appearance }));
+  }
+
+  protected updateAgendaAppearanceProvider(drop?: true): void {
+    if (!this.wantAgendaAppearanceOverride)
+      return;
+
+    for (const vp of IModelApp.viewManager) {
+      if (!this.allowView(vp))
+        continue;
+
+      if (drop || this.agenda.isEmpty)
+        vp.dropFeatureOverrideProvider(this);
+      else if (!vp.addFeatureOverrideProvider(this))
+        vp.setFeatureOverrideProviderChanged();
```

Comment:
Just fyi `vp.addFeatureOverrideProvider` already called `setFeatureOverrideProviderChanged` for you.